Mathlib/Analysis/Convex/PathSegment.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2025 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Analysis.Convex.Basic
+import Mathlib.Topology.Connected.PathConnected
+
+/-!
+# Segment between 2 points as a bundled path
+
+In this file we define `Path.segment a b : Path a b`
+to be the path going from `a` to `b` along the straight segment with constant velocity `b - a`.
+
+We also prove basic properties of this construction,
+then use it to show that a nonempty convex set is path connected.
+-/
+
+open Set
+open scoped Convex
+
+variable {E : Type*} [AddCommGroup E] [Module ℝ E]
+  [TopologicalSpace E] [ContinuousAdd E] [ContinuousSMul ℝ E]
+
+namespace Path
+
+/-- The path from `a` to `b` going along the straight segment. -/
+@[simps]
+protected def segment (a b : E) : Path a b where
+  toFun t := AffineMap.lineMap a b (t : ℝ)
+  source' := by simp
+  target' := by simp
+
+@[simp]
+theorem range_segment (a b : E) : Set.range (Path.segment a b) = [a -[ℝ] b] := by
+  simp only [segment_eq_image_lineMap, image_eq_range, ← segment_apply]
```

Comment:
Is the simp much slower if you don't squeeze it? (If not, please unsqueeze - otherwise, please add a comment. Thanks!)

Reply:
None of the lemmas from the default `simp` set apply here, so I see no reason to drop `simp`. Changed  to `rw` + `simp only` (not sure why).